rocketpy/EnvironmentAnalysis.py

```diff
@@ -281,19 +325,94 @@ def __check_coordinates_inside_grid(self, lonIndex, latIndex, lonArray, latArray
+            self.start_date = self.preferred_timezone.localize(self.start_date)
         if self.end_date.tzinfo is None:
-            self.end_date = self.prefered_timezone.localize(self.end_date)
+            self.end_date = self.preferred_timezone.localize(self.end_date)
 
-    def __find_prefered_timezone(self):
-        if self.prefered_timezone is None:
+    def __find_preferred_timezone(self):
+        if self.preferred_timezone is None:
             # Use local timezone based on lat lon pair
             tf = TimezoneFinder()
-            self.prefered_timezone = pytz.timezone(
+            self.preferred_timezone = pytz.timezone(
                 tf.timezone_at(lng=self.longitude, lat=self.latitude)
             )
-        elif isinstance(self.prefered_timezone, str):
-            self.prefered_timezone = pytz.timezone(self.prefered_timezone)
+        elif isinstance(self.preferred_timezone, str):
+            self.preferred_timezone = pytz.timezone(self.preferred_timezone)
+
+    def __init_data_parsing_units(self):
+        """Define units for pressure level and surface data parsing"""
+        self.current_units = {
+            "height_ASL": "m",
+            "pressure": "hPa",
+            "temperature": "K",
+            "windDirection": "deg",
+            "windHeading": "deg",
+            "windSpeed": "m/s",
+            "windVelocityX": "m/s",
+            "windVelocityY": "m/s",
+            "surface100mWindVelocityX": "m/s",
+            "surface100mWindVelocityY": "m/s",
+            "surface10mWindVelocityX": "m/s",
+            "surface10mWindVelocityY": "m/s",
+            "surfaceTemperature": "K",
+            "cloudBaseHeight": "m",
+            "surfaceWindGust": "m/s",
+            "surfacePressure": "Pa",
+            "totalPrecipitation": "m",
+        }
+        # Create a variable to store updated units when units are being updated
+        self.updated_units = self.current_units.copy()
+
+    def __init_unit_system(self):
+        """Initialize preferred units for output (SI, metric or imperial)."""
+        if self.unit_system_string == "metric":
+            self.unit_system = {
+                "length": "km",
+                "velocity": "m/s",
+                "acceleration": "g",
+                "mass": "kg",
+                "time": "s",
+                "pressure": "hPa",
+                "temperature": "degC",
+                "angle": "deg",
+                "precipitation": "mm",
+                "wind_speed": "m/s",
+            }
+        elif self.unit_system_string == "imperial":
+            self.unit_system = {
+                "length": "ft",
+                "velocity": "mph",
+                "acceleration": "ft/s^2",
+                "mass": "lb",
+                "time": "s",
+                "pressure": "inHg",
+                "temperature": "degF",
+                "angle": "deg",
+                "precipitation": "in",
+                "wind_speed": "knot",
```

Comment:
I would recommend changing this to windSpeed since we already have another dictionary where we currently use windSpeed as a key value and therefore having both windSpeed and wind_speed in the same context may become confuse sometimes